File: java/BubbleSort.java
/**
 * Bubble Sort is the simplest sorting algorithm that works by repeatedly swapping the adjacent
 * elements if they are in wrong order.
 */

public class BubbleSort {

  /**
   * @param numbers array of unordered integer
   * @return ordered array of numbers
   */
  int[] bubbleSort(int[] numbers) {
    for (int i = 0; i < numbers.length; i++) {
      for (int j = 0; j < numbers.length - i - 1; j++) {
        if (numbers[j] > numbers[j + 1]) {
          int temp = numbers[j];
          numbers[j] = numbers[j + 1];
          numbers[j + 1] = temp;
        }
      }
    }
    return numbers;
  }
}

File: php/bubble-sort.php
<?php

/**
 * Bubble Sort
 * 
 * Bubble Sort is the simplest sorting algorithm that works by repeatedly swapping the adjacent
 * elements if they are in wrong order. 
 * PHP Version
 * 
 */
function bubble_sort(array $array): array
{

    for ($i = 0; $i < count($array); $i++) {
        for (
            $j = 0;
            $j < count($array) - $i - 1;
            $j++
        ) {
            if ($array[$j] > $array[$j + 1]) {
                $tmp = $array[$j];
                $array[$j] = $array[$j + 1];
                $array[$j + 1] = $tmp;
            }
        }
    }
    return $array;
}
